flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -732,6 +758,18 @@ else if (current == JobStatus.RESTARTING) {
 		}
 	}
 
+	public void stop() {
+		if(jobType == JobType.STREAMING) {
+			for(ExecutionVertex ev : this.getAllExecutionVertices()) {
+				if(ev.getNumberOfInputs() == 0) { // send signal to sources only
+					ev.stop();
+				}
+			}
+		} else {
+			throw new RuntimeException("STOP is only supported by streaming jobs.");
```

Comment:
This is not a problem: in `JobManager.scala` a potential `Throwable` is caught when `ExecutionGraph.stop()` is called and a prober `StoppingFailure` message is sent back as response. I tested this in local mode. The job manager did not go down. Or is there any other place in the code in which this could become a problem? As alternative, we could introduce a regular exception (ie, `StoppingException`) and add a `throws` clause to `stop()`. What do you think about this?